pkg/api/v1/clients/kube/controller/controller.go

```diff
@@ -187,9 +197,36 @@ func (c *Controller) enqueueSync(t eventType, old, new interface{}) {
 		runtime.HandleError(err)
 		return
 	}
+	// if no relevant changes were made, no reason to queue up work
+	if reflect.DeepEqual(old, new) {
```

Comment:
Could this be a problem with very large config maps? The fact that we were hashing the `data` was causing huge CPU spikes, does `deepEquals` would compare those fields as well?

Reply:
Perhaps, if we can afford to drop events when the buffer gets full that may be a better solution. this requires more thought & testing